specs/capella/beacon-chain.md

```diff
@@ -323,9 +328,18 @@ def process_withdrawals(state: BeaconState, payload: ExecutionPayload) -> None:
     for expected_withdrawal, withdrawal in zip(expected_withdrawals, payload.withdrawals):
         assert withdrawal == expected_withdrawal
         decrease_balance(state, withdrawal.validator_index, withdrawal.amount)
-    if len(expected_withdrawals) > 0:
+
+    if len(expected_withdrawals) != 0:
         latest_withdrawal = expected_withdrawals[-1]
         state.next_withdrawal_index = WithdrawalIndex(latest_withdrawal.index + 1)
+
+    if len(expected_withdrawals) < MAX_WITHDRAWALS_PER_PAYLOAD:
+        # advance sweep if there were not a full set of withdrawals
+        next_index = state.next_withdrawal_validator_index + MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP
+        next_validator_index = next_index % len(state.validators)
+        state.next_withdrawal_validator_index = ValidatorIndex(next_validator_index)
+    else:
+        latest_withdrawal = expected_withdrawals[-1]
```

Comment:
```suggestion
```

Reply:
You need the assignment in there or an outer assignment beacuse the prior if isn't necessarily hit. I cover that in my comment below

Reply:
> You need the assignment in there or an outer assignment beacuse the prior if isn't necessarily hit. I cover that in my comment below

no because this else is only hit when `len(expected_withdrawals) >= MAX_WITHDRAWALS_PER_PAYLAOD`  in which case the previous assignment already was hit.

Reply:
so I'm trying to avoid python's dynamic scoping here and so just elected to make the assignment twice

if we are ok w/ imo non-standard (relative to what most readers are assumed to know) scoping behavior then we can drop the extra assignment